schema/InlineContent.schema.yaml

```diff
@@ -31,5 +31,7 @@ anyOf:
   - type: integer
   - type: number
   - type: array
+    items:
+      $ref: InlineContent
   - type: object
```

Comment:
Having `object` present as an allowed `InlineContent` element doesn't sit well with me, because `object` essentially translates to allowing any `Entity` being accepted. If you're ok with it @nokome, I'd like to remove if from the list.

Array isn't ideal in my mind, but acceptable with the constraint that all children must also conform to the `InlineContent` schema.

I understand that this section is in the context of coercion, but it seems to me counter to the purpose of defining a strict schema, and instead couples it to the codec implementation to know that it must filter/coerce any non-inline elements.

Open to opposing thoughts though :)

Reply:
I agree that having `Object` and/or `Array` as members of `InlineContent` is problematic. The main use case for this that I see is to have the `output` of a `CodeExpression` be a lists e.g.

```json
{
  "type": "CodeExpression",
  "text": "favouriteFruit.slice(0,3)",
  "output": ["apple", "pear", "orange"]
}
```

Which I might want to place in a paragraph and have the output rendered as a comma separated list e.g. The top three fruit reported in the district were: apple, pear, orange.

With this in mind, I think I added `Object` and `Array` (back in?) as members of `InlineContent` ahead of wanting to narrow the `output` of a `CodeExpression` for `Node` to `InlineContent` - i.e. a `CodeExpression` should only ever output something that is valid `InlineContent`. However, I suspect it is better to let both `CodeExpression` and `CodeChunk` have any `Node` as output and instead deal with how to represent them as inline content (or block content) when it comes to encoding in each format.

Constraining an array to `InlineContent` doesn't feel like a good solution: it is confusingly circular (should that array get flattened if it is part of an `Array<InlineContent>`?).

So, in the end, I have removed both `Object` and `Array` from `InlineContent`.